source/Examples/SmartCache/Main.cpp

```diff
@@ -99,37 +161,72 @@ main(
             CyclicalWorkload(2048, smartCache);
         }
 
+        // After having run a workload for a while, we want to check for a new
+        // config suggestion from an optimizer.
+        // In this case we make it a blocking call.
+
+        // First, create some condition variables to help signal when the new
+        // config is ready to be consumed.
+        //
         bool isConfigReady = false;
         std::mutex waitForConfigMutex;
         std::condition_variable waitForConfigCondVar;
 
-        // Setup a callback.
+        // Also, setup a callback lambda function for handling the
+        // SharedConfigUpdatedFeedbackeMessage we expect to receive from the
+        // agent after we request a config update with a RequestNewConfigurationMessage.
+        //
+        // Note: this lambda will be invoked by the background task setup above
+        // for processing messages on the feedback channel.
         //
         ObjectDeserializationCallback::Mlos::Core::SharedConfigUpdatedFeedbackMessage_Callback =
             [&waitForConfigMutex, &waitForConfigCondVar,
              &isConfigReady](Proxy::Mlos::Core::SharedConfigUpdatedFeedbackMessage&& msg)
             {
-                // Ignore the message.
+                // The contents of the message are irrelevant in this case.
+                // It's just a signal that the RequestNewConfigurationMessage
+                // has been processed by the agent.
                 //
                 UNUSED(msg);
 
+                // So, we will just notify the waiting loop (below) that the
+                // message has been processed now and is ready to be read.
+                //
                 std::unique_lock<std::mutex> lock(waitForConfigMutex);
                 isConfigReady = true;
                 waitForConfigCondVar.notify_all();
             };
 
-        // Send a request to obtain a new configuration.
+        // Send a request to obtain a new configuration from the optimizer.
+        //
+        // Note: the message (as defined in
+        // SmartCache.SettingsRegistry/CodeGen/SmartCache.cs) has no members, so
+        // there's no details to fill in here (and it is just zero-initialized).
+        // It's simply a signal to send to the external agent to request a new
+        // config be populated in the shared memory region.
         //
         SmartCache::RequestNewConfigurationMessage msg = { 0 };
         mlosContext.SendTelemetryMessage(msg);
 
+        // Now, we wait for the external agent to respond to our request.
```

Comment:
If no optimizer is registered then... ? it'll send a message without touching the shared memory?

Reply:
There's a comment about that in `SmartCache.SettingsRegistry/AssemblyInitializer.cs`.

Basically, if there's no optimizer, the request is a no-op and just continues to use the existing value in shared memory.

Reply:
Do you think it needs mention of that here as well?

Reply:
added another comment to help them track the flow to the RequestNewConfigurationMessage handler in that AssemblyInitializer.cs file